server/datastore/mysql/hosts.go

```diff
@@ -938,10 +923,8 @@ func (ds *Datastore) EnrollHost(ctx context.Context, osqueryHostID, nodeKey stri
 			}
 			hostID = int64(host.ID)
 
-			if shouldCleanTeamPolicies(host.TeamID, teamID) {
-				if err := cleanupPolicyMembershipOnTeamChange(ctx, tx, []uint{host.ID}); err != nil {
-					return ctxerr.Wrap(ctx, err, "EnrollHost delete policy membership")
-				}
+			if err := deleteAllPolicyMemberships(ctx, tx, []uint{host.ID}); err != nil {
+				return ctxerr.Wrap(ctx, err, "cleanup policy membership on re-enroll")
```

Comment:
@chiiph I'm thinking it might be worth running a load test for this (it will require a bit of tweaking to osquery-perf to make hosts re-enroll after a period of time)

any thoughts?

Reply:
Good idea.

> it will require a bit of tweaking to osquery-perf to make hosts re-enroll after a period of time

We should add some support for osquery-perf to reenroll when it gets an invalid node key error back from Fleet. Then one could test a re-enroll thundering herd (not sure how often this scenario happens on production) by deleting ALL hosts from Fleet. IMO this kind of load testing should not block this PR though.

Reply:
Sounds good to me, osquery-perf should do as much as osquery does in the ideal world.